api/app/signals/apps/signals/models/category_assignment.py

```diff
@@ -32,6 +32,12 @@ class CategoryAssignment(CreatedUpdatedModel):
     deadline = models.DateTimeField(null=True)
     deadline_factor_3 = models.DateTimeField(null=True)
 
+    # SIG-3555 store handling message as it was at the moment the category
+    # was assigned. Before we had problems with history "changing" when the
+    # handling message for a category was updated (because it was queried from
+    # the category table each time the history is shown).
+    stored_handling_message = models.TextField(null=True)
```

Comment:
I like this solution, it will tell you what the handling_message is at the moment of (initial creating) updating the category assignment of a Signal.

However this is not a "history" for the handling_message itself. So if that is something we want to support this will not do that.

Reply:
Agree, the history is created in the history (SQL)view. What this does is store the then current `handling_message` at each category update. Strictly speaking we only need the first one for now, but having them for all updates gives us some flexibility if business rules change. If these change we wil have a dataset where for each signal and each category update there is a correct stored handling message (that cannot be reconstructed otherwise --- since the Category model does not fully track history of handling message updates).

Reply:
As discussed, we can leave this in the `save` for now and later do a refactor that cleans up all model `saves`.